thewalrus/libwalrus.pyx

```diff
@@ -468,5 +573,185 @@ def renorm_hermite_multidimensional_real(double [:, :] R, double [:] y, int cuto
 
     for i in range(n):
         y_mat.push_back(y[i])
+    length = cutoff**n
+    cdef double *array = renorm_hermite_multidimensional_cpp(R_mat, y_mat, cutoff)
+
+    return double_pointer_to_array(array, length)
+
+
+def interferometer(double complex[:, :] R, int cutoff):
+    r"""Returns the matrix elements of an interferometer by using a custom version of
+    the renormalized Hermite polynomials.
+
+    Args:
+        R (array[complex128]): square matrix parametrizing the interferometer
+        cutoff (int): maximum size of the subindices in the tensor
+
+    Returns:
+        array[complex128]: the matrix elements of the interferometer
+    """
+    cdef int i, j, n = R.shape[0]
+    cdef vector[double complex] R_mat
+
+    for i in range(n):
+        for j in range(n):
+            R_mat.push_back(R[i, j])
+
+    length = cutoff**n
+    cdef double complex *array = interferometer_cpp(R_mat, cutoff)
+
+    return np.reshape(complex_pointer_to_array(array, length), [cutoff]*n)
+
+def interferometer_real(double [:, :] R, int cutoff):
+    r"""Returns the matrix elements of an interferometer by using a custom version of
+    the renormalized Hermite polynomials.
+
+    Args:
+        R (array[float64]): square matrix parametrizing the interferometer
+        cutoff (int): maximum size of the subindices in the tensor
+
+    Returns:
+        array[float64]: the matrix elements of the interferometer
+    """
+    cdef int i, j, n = R.shape[0]
+    cdef vector[double] R_mat
+
+    for i in range(n):
+        for j in range(n):
+            R_mat.push_back(R[i, j])
+    length = cutoff**n
+    cdef double *array = interferometer_cpp(R_mat, cutoff)
+
+    return np.reshape(double_pointer_to_array(array, length), [cutoff]*n)
+
+
+def squeezing(double complex[:, :] R, int cutoff):
+    r"""Returns the matrix elements of a single mode squeezer by using a custom version of
+    the renormalized Hermite polynomials.
+
+    Args:
+        R (array[complex128]): square matrix parametrizing the squeezing
+        cutoff (int): maximum size of the subindices in the tensor
+
+    Returns:
+        array[complex128]: the matrix elements of the squeezing operator
+    """
+    cdef int i, j, n = 2
+    cdef vector[double complex] R_mat
+
+    for i in range(n):
+        for j in range(n):
+            R_mat.push_back(R[i, j])
+    length = cutoff**n
+    cdef double complex *array = squeezing_cpp(R_mat, cutoff)
+
+    return np.reshape(complex_pointer_to_array(array, length), [cutoff]*n)
+
+
+def squeezing_real(double [:, :] R, int cutoff):
+    r"""Returns the matrix elements of a single mode squeezer by using a custom version of
+    the renormalized Hermite polynomials.
+
+    Args:
+        R (array[float64]): square matrix parametrizing the squeezing

```

Comment:
still not really sure what this is :thinking: Is it the covariance matrix?

Reply:
See above. It is not the covariance matrix.